jumpgate/volume/drivers/sl/volumes.py

```
@@ -295,25 +297,31 @@ def _match_portable_storage_prices(packageId, size, exact_capacity):
             price_matrix = {}
             for x in price_list:
                 price_matrix.update({int(x['capacity']): x['prices']})
-            # find the closet capacity to the requested size
-            if exact_capacity:
-                ret = False
-                for x in price_matrix:
-                    if size == x:
-                        ret = True
-                        capacity_idx = x
-                        break
-                if not ret:
-                    raise SoftLayer.SoftLayerAPIError(
-                        HTTP.BAD_REQUEST,
-                        'volume_types: extra_specs: '
-                        'drivers:exact_capacity is set to'
-                        ' True and there is no volume with'
-                        ' matching capacity')
-            else:
-                capacity_idx = min(price_matrix, key=lambda x: abs(x - size))
-
-            return price_matrix[capacity_idx]
+            try:
+                # find the closet capacity to the requested size
+                if exact_capacity:
+                    ret = False
+                    for x in price_matrix:
+                        if size == x:
+                            ret = True
+                            capacity_idx = x
+                            break
+                    if not ret:
+                        raise SoftLayer.SoftLayerAPIError(
+                            HTTP.BAD_REQUEST,
+                            'volume_types: extra_specs: '
+                            'drivers:exact_capacity is set to'
+                            ' True and there is no volume with'
+                            ' matching capacity')
+                else:
+                    capacity_idx = sorted([cap for cap in price_matrix
+                                           if cap >= size])[0]
+
+                return price_matrix[capacity_idx]
+            except IndexError:
```

Any reason this try/except shouldn't wrap only the code at lines 317-318, rather than the whole if/else block? Looks like that is the only code here that should raise this exception.